src/definition/livechat/ILivechatRoom.ts

```diff
@@ -29,7 +29,7 @@ type OmnichannelSource =
     | IOmnichannelSourceApp;
 
 export interface ILivechatRoom extends IRoom {
-    visitor: IVisitor;
+    visitor: IVisitor & { lastMessageTs?: Date };
```

Comment:
I'm a bit confused on why you've introduced the new property with the intersection operator here instead of in `IVisitor` itself, can you clarify?

Reply:
Hi @d-gubert I share more context about this over [here](https://github.com/RocketChat/Rocket.Chat/pull/29756/files#r1257265249). 
In-short: this additional prop is stored on room level since we can have a single visitor contacting us using multiple channels, hence we're storing all channel related info on room level. LastMessageTs is one of such channel level prop.

Reply:
Thanks for the feedback Doug. I've updated the PR with your suggestions